### PR TITLE
tests: test gk-deploy syntax

### PR DESCRIPTION
**Pull request by obnoxxx**
Signed-off-by: Michael Adam <obnox@redhat.com>

---
This change is [Reviewable](https://reviewable.io/reviews/gluster/gluster-kubernetes/213)

### CODE REVIEW

**Comment by obnoxxx** [comment]
This tests the syntactic correctness of the gk-deploy script.

It would have caught the error of #208 like this:

```
...
test: test script syntax
time: 2017-03-26 20:43:42.994495Z
failure: test script syntax [
/data/obnox/devel/gluster/gluster-kubernetes/tests/gk-deploy/../../deploy/gk-deploy: line 492: conditional binary operator expected
/data/obnox/devel/gluster/gluster-kubernetes/tests/gk-deploy/../../deploy/gk-deploy: line 492: syntax error near `"${CLI}"'
/data/obnox/devel/gluster/gluster-kubernetes/tests/gk-deploy/../../deploy/gk-deploy: line 492: `  if [[ "${heketi_service}" != "<none>"]] && [[ "${CLI}" == *oc\ * ]]; then'
]
...
```

**Comment by obnoxxx** [comment]
@MohamedAshiqrh ptal

**Comment by obnoxxx** [comment]
@jarrpa

**Comment by jarrpa** [comment]
Could we try out and (if we like it) conditionally make use of ShellCheck if it's installed?

https://www.shellcheck.net/

It seems that it is supported in Travis CI:

https://github.com/koalaman/shellcheck#user-content-installing

**Comment by obnoxxx** [comment]
Never heard of shellcheck before... experimenting with it right now

**Comment by jarrpa** [comment]
Sorry, wrong heading in the page. You want to modify the travis config like so:
```
addons:
  apt:
    sources:
    - debian-sid    # Grab ShellCheck from the Debian repo
    packages:
    - shellcheck
```
https://github.com/koalaman/shellcheck#travis-ci-setup

**Comment by obnoxxx** [comment]
yeah, there seem to be different possibilities. here is another one:
https://www.sysorchestra.com/2016/07/18/introduction-on-how-to-use-shellcheck-and-bashate-with-travis-ci/
not using sid but backports

**Comment by jarrpa** [comment]
I'd sooner trust backports, yeah. :)

File: tests/README.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ Install it with e.g.
 * `dnf install yamllint`, or
 * `pip install yammling`
 
+The gk-deploy test uses ShellCheck inf installed.
```
Comment (jarrpa) [inline tests/README.md]: Minot typo detected: `inf` should be `if`.

Comment (obnoxxx) [inline tests/README.md]: done

**Comment by jarrpa** [comment]
LGTM

**Pull request merged**